src/runtime/topology.rs

```diff
@@ -118,11 +119,18 @@ impl Topology {
     }
 
     /// Adds a [Block] to the [Topology] returning the `id` of the [Block] in the [Topology].
-    pub fn add_block(&mut self, mut block: Block) -> usize {
-        let block_name = block.type_name();
-        let block_id = self.blocks.vacant_key();
-        block.set_instance_name(format!("{}-{}", block_name, block_id));
-        self.blocks.insert(Some(block))
+    pub fn add_block(&mut self, mut block: Block) -> Result<usize> {
```

Comment:
This should be a normal `Result<usize, crate::runtime::Error>`, not an `anyhow::Result`.

Reply:
Good catch... I've not been paying attention to the "error" contexts as I should.